File: CS/DXSample/SchedulingViewModel.cs
using System;
using System.Collections.ObjectModel;
using DevExpress.Mvvm;

namespace DXSample.ViewModels {
    public class SchedulingViewModel : ViewModelBase {
        public SchedulingViewModel() {
            Appointments = DataHelper.GetAppointments();
            Resources = DataHelper.GetResources();
        }
        public ObservableCollection<AppointmentViewModel> Appointments {
            get { return GetValue<ObservableCollection<AppointmentViewModel>>(); }
            set { SetValue(value); }
        }
        public ObservableCollection<ResourceViewModel> Resources {
            get { return GetValue<ObservableCollection<ResourceViewModel>>(); }
            set { SetValue(value); }
        }
    }
    public class AppointmentViewModel : ViewModelBase {
        public int Id {
            get { return GetValue<int>(); }
            set { SetValue(value); }
        }
        public DateTime StartTime {
            get { return GetValue<DateTime>(); }
            set { SetValue(value); }
        }
        public DateTime EndTime {
            get { return GetValue<DateTime>(); }
            set { SetValue(value); }
        }
        public string Subject {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
        public int ResourceId {
            get { return GetValue<int>(); }
            set { SetValue(value); }
        }
    }
    public class ResourceViewModel : ViewModelBase {
        public int Id {
            get { return GetValue<int>(); }
            set { SetValue(value); }
        }        
        public string Caption {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
    }
}

File: VB/DXSample/SchedulingViewModel.vb
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm

Namespace DXSample.ViewModels

    Public Class SchedulingViewModel
        Inherits ViewModelBase

        Public Sub New()
            Appointments = DataHelper.GetAppointments()
            Resources = DataHelper.GetResources()
        End Sub

        Public Property Appointments As ObservableCollection(Of AppointmentViewModel)
            Get
                Return GetValue(Of ObservableCollection(Of AppointmentViewModel))()
            End Get

            Set(ByVal value As ObservableCollection(Of AppointmentViewModel))
                SetValue(value)
            End Set
        End Property

        Public Property Resources As ObservableCollection(Of ResourceViewModel)
            Get
                Return GetValue(Of ObservableCollection(Of ResourceViewModel))()
            End Get

            Set(ByVal value As ObservableCollection(Of ResourceViewModel))
                SetValue(value)
            End Set
        End Property
    End Class

    Public Class AppointmentViewModel
        Inherits ViewModelBase

        Public Property Id As Integer
            Get
                Return GetValue(Of Integer)()
            End Get

            Set(ByVal value As Integer)
                SetValue(value)
            End Set
        End Property

        Public Property StartTime As Date
            Get
                Return GetValue(Of Date)()
            End Get

            Set(ByVal value As Date)
                SetValue(value)
            End Set
        End Property

        Public Property EndTime As Date
            Get
                Return GetValue(Of Date)()
            End Get

            Set(ByVal value As Date)
                SetValue(value)
            End Set
        End Property

        Public Property Subject As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public Property ResourceId As Integer
            Get
                Return GetValue(Of Integer)()
            End Get

            Set(ByVal value As Integer)
                SetValue(value)
            End Set
        End Property
    End Class

    Public Class ResourceViewModel
        Inherits ViewModelBase

        Public Property Id As Integer
            Get
                Return GetValue(Of Integer)()
            End Get

            Set(ByVal value As Integer)
                SetValue(value)
            End Set
        End Property

        Public Property Caption As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property
    End Class
End Namespace
